cmd/cloudshell_open/main.go

```diff
@@ -250,48 +250,54 @@ func run(opts runOpts) error {
 		return err
 	}
 
-	exists, err := dockerFileExists(appDir)
-	jibMaven := false
-	if err != nil {
-		return err
+	projectEnv := fmt.Sprintf("GOOGLE_CLOUD_PROJECT=%s", project)
+	regionEnv := fmt.Sprintf("GOOGLE_CLOUD_REGION=%s", region)
+	serviceEnv := fmt.Sprintf("K_SERVICE=%s", serviceName)
+	imageEnv := fmt.Sprintf("IMAGE_URL=%s", image)
+	appDirEnv := fmt.Sprintf("APP_DIR=%s", appDir)
+	pathEnv := fmt.Sprintf("PATH=%s", os.Getenv("PATH"))
+
+	hookEnvs := append([]string{projectEnv, regionEnv, serviceEnv, imageEnv, appDirEnv, pathEnv}, envs...)
+
+	pushImage := true
+
+	if appFile.Hooks.PreBuild.Commands != nil {
+		err = runScripts(appDir, appFile.Hooks.PreBuild.Commands, hookEnvs)
 	}
-	if exists {
-		fmt.Println(infoPrefix + " Attempting to build this application with its Dockerfile...")
-		fmt.Println(infoPrefix + " FYI, running the following command:")
-		cmdColor.Printf("\tdocker build -t %s %s\n", parameter(image), parameter("."))
-	} else {
-		jibMaven, err = jibMavenConfigured(appDir)
-		if err != nil {
-			return fmt.Errorf("failed to check if Jib is configured: %s", err)
-		}
-		if jibMaven {
+
+	if appFile.Build.Skip == nil || *appFile.Build.Skip == false {
```

Comment:
lets rewrite this as to avoid extra indentation:

```
if (skip condition) {
    continue
} else if dockerfile {
   ...
} else if jib {
   ...
} ...
```